arangod/Aql/Query.cpp

```diff
@@ -149,6 +149,11 @@ Query::Query(std::shared_ptr<transaction::Context> const& ctx,
 
   _resourceMonitor.setMemoryLimit(_queryOptions.memoryLimit);
   _warnings.updateOptions(_queryOptions);
+
+  // store name of user that started the query
+  if (!ServerState::instance()->isDBServer()) {
+    _user = ExecContext::current().user();
```

Comment:
User is empty anyway no need to check

Reply:
Why? If this is running on coordinator or single server, we _should_ have a user set in most case.

Reply:
or are you suggesting we should assign to the variable unconditionally?